tiledb/tests/test_libtiledb.py

```diff
@@ -3206,12 +3209,41 @@ def write_fragments(target_path):
         self.assertEqual(fi.unconsolidated_metadata_num, num_writes)
 
         conf = tiledb.Config({"sm.consolidation.mode": "fragment_meta"})
-        with tiledb.open(path3, "w") as A:
-            A.consolidate(config=conf)
+        tiledb.consolidate(uri=path3, config=conf)
 
         fi = tiledb.array_fragments(path3)
         self.assertEqual(fi.unconsolidated_metadata_num, 0)
 
+    def test_array_consolidate_read_mode(self):
+
+        dshape = (1, 3)
+        num_writes = 10
+
+        def create_array(target_path, dshape):
+            dom = tiledb.Domain(tiledb.Dim(domain=dshape, tile=len(dshape)))
+            att = tiledb.Attr(dtype="int64")
+            schema = tiledb.ArraySchema(domain=dom, attrs=(att,), sparse=True)
+            tiledb.libtiledb.Array.create(target_path, schema)
+
+        def write_fragments(target_path, dshape, num_writes):
+            for i in range(1, num_writes + 1):
+                with tiledb.open(target_path, "w") as A:
+                    A[[1, 2, 3]] = np.random.rand(dshape[1])
+
+        path = self.path("test_array_consolidate_read_mode")
+
+        create_array(path, dshape)
+        write_fragments(path, dshape, num_writes)
+
+        config = tiledb.Config()
+
+        ctx = tiledb.Ctx(config=config)
+        lt_array = lt.Array(ctx, path, lt.QueryType.READ)
```

Comment:
Ah ok I did not catch this before. The underlying `tiledb.cc` (`lt`) bindings are not meant to be public. The user should still be using `tiledb.Array`. [For example](https://github.com/TileDB-Inc/TileDB-Py/blob/agis/sc-44909/move-fragment-list-consolidation-api-to-pybind/tiledb/array_schema.py#L36-L75), `tiledb.ArraySchema` inherits from `lt.Schema`. And this way we can set the default ctx to `tiledb.default_ctx()`.

Reply:
What I think is that the static `tiledb.consolidate` method can be in Pybind11, but we still need the Cython  `tiledb.libtiledb.Array.consolidate` method. Otherwise, to create a pure Python `Array` class as we did for `ArraySchema`, you would need to move everything in the `Array` class out of Cython and into Pybind11 which is a TON of work.

Reply:
That's exactly the problem. That's the reason that it wasn't clear what to do with that story.